src/components/VisibilityContainer/index.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useRef, useEffect, useState } from "react";
+
+interface VisibilityContainerProps {
+  children: React.ReactNode;
+}
+/**
+ * `VisibilityContainer` is a component that will only render its children when it is visible in the viewport.
+ */
+const VisibilityContainer: React.VFC<VisibilityContainerProps> = ({
+  children,
+}) => {
+  const containerRef = useRef(null);
+  const [isVisible, setIsVisible] = useState(false);
+
+  useEffect(() => {
+    const currentRef = containerRef.current;
+    const observer = new IntersectionObserver(([entry]) => {
+      setIsVisible(entry.isIntersecting);
+    });
+    observer.observe(currentRef);
+    return () => observer.unobserve(currentRef);
```

Comment:
can we `disconnect` or does it need to be `unobserve`?

Reply:
Sure, I don't see much specific guidance on which one I should use so disconnect works for me as well.